querydsl-sql/src/test/java/com/querydsl/sql/SelectBase.java

```diff
@@ -142,6 +141,8 @@ public void Arithmetic() {
         NumberExpression<Integer> three = Expressions.numberTemplate(Integer.class, "(3.0)");
         NumberExpression<Integer> four = Expressions.numberTemplate(Integer.class, "(4.0)");
         arithmeticTests(one, two, three, four);
+        // the following one doesn't work with integer arguments
+        assertEquals(2, firstResult(four.multiply(one.divide(two))).intValue());
```

Comment:
the results are consistent over databases, and the difference comes from missing normalization across operation boundaries.